gateway/src/commands/mod.rs

```diff
@@ -50,82 +86,72 @@ fn parse_validators(raw: &str) -> Vec<Url> {
-            // if port was overridden, it must be parsable
-            panic!("Invalid port value provided - {:?}", err);
-        }
-        config = config.with_mix_port(mix_port.unwrap());
+    if let Some(mix_port) = args.mix_port {
+        config = config.with_mix_port(mix_port);
     }
 
-    if let Some(clients_port) = matches
-        .value_of(CLIENTS_PORT_ARG_NAME)
-        .map(|port| port.parse::<u16>())
-    {
-        if let Err(err) = clients_port {
-            // if port was overridden, it must be parsable
-            panic!("Invalid port value provided - {:?}", err);
-        }
-        config = config.with_clients_port(clients_port.unwrap());
+    if let Some(clients_port) = args.clients_port {
+        config = config.with_clients_port(clients_port);
     }
 
-    if let Some(announce_host) = matches.value_of(ANNOUNCE_HOST_ARG_NAME) {
+    if let Some(announce_host) = args.announce_host {
         config = config.with_announce_address(announce_host);
     } else if was_host_overridden {
         // make sure our 'mix-announce-host' always defaults to 'mix-host'
-        config = config.announce_host_from_listening_host()
+        config = config.announce_host_from_listening_host();
     }
 
-    if let Some(raw_validators) = matches.value_of(VALIDATOR_APIS_ARG_NAME) {
-        config = config.with_custom_validator_apis(parse_validators(raw_validators));
-    }
-
-    #[cfg(not(feature = "coconut"))]
-    if let Some(raw_validators) = matches.value_of(VALIDATORS_ARG_NAME) {
-        config = config.with_custom_validator_nymd(parse_validators(raw_validators));
-    } else {
-        config = config.with_custom_validator_nymd(parse_validators(DEFAULT_VALIDATOR_ENDPOINT));
+    if let Some(raw_validators) = args.validator_apis {
+        config = config.with_custom_validator_apis(parse_validators(&raw_validators));
     }
 
-    #[cfg(not(feature = "coconut"))]
-    if let Some(cosmos_mnemonic) = matches.value_of(COSMOS_MNEMONIC) {
-        config = config.with_cosmos_mnemonic(String::from(cosmos_mnemonic));
-    } else {
-        config = config.with_cosmos_mnemonic(String::from(DEFAULT_MNEMONIC));
+    if let Some(wallet_address) = args.wallet_address {
+        let trimmed = wallet_address.trim();
+        validate_bech32_address_or_exit(trimmed);
+        config = config.with_wallet_address(trimmed);
     }
 
-    if let Some(datastore_path) = matches.value_of(DATASTORE_PATH) {
+    if let Some(datastore_path) = args.datastore {
         config = config.with_custom_persistent_store(datastore_path);
     }
 
-    #[cfg(not(feature = "coconut"))]
-    if let Some(eth_endpoint) = matches.value_of(ETH_ENDPOINT) {
-        config = config.with_eth_endpoint(String::from(eth_endpoint));
-    } else {
-        config = config.with_eth_endpoint(String::from(DEFAULT_ETH_ENDPOINT));
-    }
+    // Set the test mode to always on, except if build with `feature = eth`, then we check the flag
+    config = config.with_testnet_mode(true);
```

Comment:
I think you forgot to remove that one, because just underneath it, it's properly locked behind the feature flag : )

Reply:
Currently we do 
```rust
    if !cfg!(feature = "eth") || matches.is_present(TESTNET_MODE_ARG_NAME) {
        config.with_testnet_mode(true)
```
so the above line is to cover for when `not(feature = eth)`. That said, I just tried to match the old behaviour, perhaps we want to change this?

Reply:
But yes you're right, this currently logic is fishy, I'm redoing it. And good tip about generating the config template to confirm it all works! (it didn't...)